src/node_util.cc

```diff
@@ -21,17 +21,20 @@ static void GetOwnNonIndexProperties(
   Environment* env = Environment::GetCurrent(args);
   Local<Context> context = env->context();
 
-  if (!args[0]->IsObject())
+  if (!args[0]->IsObject() || !args[1]->IsUint32())
     return;
 
-  v8::Local<v8::Object> object = args[0].As<v8::Object>();
+  Local<Object> object = args[0].As<Object>();
+
+  Local<Array> properties;
 
-  // Return only non-enumerable properties by default.
-  v8::Local<v8::Array> properties;
+  v8::PropertyFilter filter =
+    static_cast<v8::PropertyFilter>(
+      args[1]->Uint32Value(env->context()).FromJust());
```

Comment:
Since you already know at this point that `args[1]` is an Uint32 here, I’d suggest using `args[1].As<Uint32>()->Value()` (which is a static cast that fully compiles away, whereas `Uint32Value()` performs the equivalent of `value|0` in JS, i.e. performs a real conversion that can potentially run userland code for non-integer input)